### PR TITLE
Replace image proxy, add auto-fetch, add Dockerfile

### PR DESCRIPTION
**Pull request by kylerm42**
I was interested in using this, and changed/added a few things that may make it easier and/or more usable for others.

- [Open Library](https://openlibrary.org/developers/api) has a public API that can supply cover images. Since the Google Books API provides an ISBN, we can use Open Library to fetch the cover image, rather than using your image proxy. This lets us remove one of the security concerns listed in the Readme.
- Speaking of security concerns, I've added an environment variable that can toggle the auto-update functionality. Against my instincts, I kept it on by default, although I'd much prefer it being opt-in rather than opt-out.
- I added a way for the app to automatically fetch on a specified interval, rather than requiring the `/fetch` endpoint to be called. The `AUTO_FETCH_INTERVAL` environment variable specifies how long to wait (in ms) between automatically checking for new pages.
- There is also a new Dockerfile, so people can easily host this locally or wherever they want. Instructions for this were added to the Readme.

Some of these changes would also require changes to the guide you have as well.

### CODE REVIEW

File: modules.js [hunk 1]
```diff
@@ -6,183 +6,173 @@ const { Client } = require("@notionhq/client");
 
 require("dotenv").config();
 
-if (!(process.env.NOTION_API_KEY && process.env.DATABASE_ID)) {
-    throw new Error("Please fill in your API key and database ID in repl.it");
-}
```
Comment (kylerm42) [inline modules.js:11]: I moved this to the main function when starting the app. It seemed like an app critical piece that fit better over there.

File: modules.js [hunk 2]
```diff
@@ -6,183 +6,173 @@ const { Client } = require("@notionhq/client");
...
-  
-    const databaseId = process.env.DATABASE_ID;
+    let updateOptions = {
+      page_id: i.id,
 
-    const queryResponse = await notion.databases.query({
-        database_id: databaseId,
-        page_size: 100,
-        filter: {
-          or: [
```
Comment (kylerm42) [inline modules.js:29]: This `or` seemed unnecessary with only one option provided.

File: modules.js [hunk 3]
```diff
@@ -6,183 +6,173 @@ const { Client } = require("@notionhq/client");
...
-                number: book.volumeInfo.pageCount,
-            };
-        }
-
-        try {
+          ],
+        },
+
+        Link: {
+          url: `https://openlibrary.org/isbn/${isbn}`,
```
Comment (kylerm42) [inline modules.js:110]: Open Library asks for a link back if using their [covers API](https://openlibrary.org/dev/docs/api/covers), so this seemed like a harmless change for their service.

**Comment by shaunakg** [comment]
Nice! Thank you for these changes, I will merge them when I've rewritten the guide (shouldn't take more than a couple of hours). 

Regarding the auto-update functionality - you are right about this and I felt uneasy when implementing it, but I needed a way to deploy updates to servers on Replit not under my control because there were a lot of API degradations that I had to address in the past few weeks. So what I'll do is make it so that it only notifies when there's a new version available, perhaps through the notion page itself (by editing the description maybe). This would mitigate the risk.

**Comment by shaunakg** [comment]
I think I have a better auto-update functionality, awaiting testing. This will create a page in the notion database notifying users that there's an update, and they can just tick a checkbox in order to pull the new code. And it also respects the environment variable in regards to checking for updates.

**Comment by kylerm42** [comment]
> I think I have a better auto-update functionality, awaiting testing. This will create a page in the notion database notifying users that there's an update, and they can just tick a checkbox in order to pull the new code. And it also respects the environment variable in regards to checking for updates.

Oh that's smart!

**Pull request merged**